pkg/drivers/kic/oci/info.go

```diff
@@ -225,29 +225,41 @@ type podmanSysInfo struct {
 }
 
 // dockerSystemInfo returns docker system info --format '{{json .}}'
-func dockerSystemInfo() (dockerSysInfo, error) {
+func dockerSystemInfo(mockTest ...string) (dockerSysInfo, error) {
 	var ds dockerSysInfo
-	rr, err := runCmd(exec.Command(Docker, "system", "info", "--format", "{{json .}}"))
-	if err != nil {
-		return ds, errors.Wrap(err, "get docker system info")
+	var rawJson string
+	if len(mockTest) > 0 {
+		rawJson = mockTest[0]
+	} else {
+		rr, err := runCmd(exec.Command(Docker, "system", "info", "--format", "{{json .}}"))
+		if err != nil {
+			return ds, errors.Wrap(err, "get docker system info")
+		}
+		rawJson = rr.Stdout.String()
 	}
 
-	if err := json.Unmarshal([]byte(strings.TrimSpace(rr.Stdout.String())), &ds); err != nil {
+	if err := json.Unmarshal([]byte(strings.TrimSpace(rawJson)), &ds); err != nil {
 		return ds, errors.Wrapf(err, "unmarshal docker system info")
 	}
 
 	return ds, nil
 }
 
 // podmanSysInfo returns podman system info --format '{{json .}}'
-func podmanSystemInfo() (podmanSysInfo, error) {
+func podmanSystemInfo(mockTest ...string) (podmanSysInfo, error) {
```

Comment:
Instead of adding an argument to many functions, I would create a function that can be overwritten in the unit test itself and return whatever data you want:


```golang
var (
  GetDockerInfoOutput = dockerSystemInfo
)

```

and in the unit test:

```go
GetDockerInfoOutput = func() (dockerSysInfo, error) {
  return rawJson, nil
}
```

Reply:
I like this idea, seems like an interesting mocking way, but I also think this one might be easier to maintain with less complexity. but I could refactor to do that if you think that would add more value. wdyt?

Reply:
I definitely feel this way is cleaner than having to pass in arguments to multiple functions -- that could get complex quickly if you end up having to write more functions

Reply:
done